counsel-ag-popup.el

```diff
@@ -0,0 +1,137 @@
+;;; counsel-ag-popup.el --- Interactive search with ag -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2019  Eder Elorriaga
+
+;; Author: Eder Elorriaga <gexplorer8@gmail.com>
+;; Keywords: convenience, matching, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Interactive search with ag.
+;; To enable, put this code in your init file:
+;; (use-package counsel-ag-popup
+;;  :after counsel
+;;  :bind ("C-c s" . counsel-ag-popup))
```

Comment:
I don't think we can assume every user uses `use-package`. In general, each package needn't explain how to install and activate Emacs packages - that's what the Emacs manual and built-in help system are for. The only thing packages ought to describe in their `Commentary` is their entrypoints - which commands are most pertinent, and what they do. The rest, like `eval-after-load` and key binding, are general Emacs topics, not package-specific ones.